packages/wallet-adapter-react/src/WalletProvider.tsx

```diff
@@ -74,29 +75,43 @@ export const AptosWalletAdapterProvider: FC<AptosWalletProviderProps> = ({
     }
   };
 
-  const signAndSubmitTransaction = async (
-    transaction: Types.TransactionPayload
+  const signAndSubmitTransaction = async <
+    T extends Types.TransactionPayload,
+    V extends TransactionOptions | undefined
```

Comment:
I'm not a big typescript dev, so I don't know.  Is this standard practice to provide types that can be extended?

Reply:
it is better for flexibility and extensibility while making sure the base type is used and satisfied

Reply:
Thank you for your valuable suggestions. commit  [1281dba](https://github.com/aptos-labs/aptos-wallet-adapter/pull/156/commits/1281dba03db8c0e8140c1339573a27fd636c287a)